astropy/cosmology/flrw.py

```diff
@@ -2877,9 +2872,9 @@ def __init__(self, H0, Om0, Ode0, wp=-1.0, wa=0.0, zp=0.0, Tcmb0=0.0*u.K,
                  Neff=3.04, m_nu=0.0*u.eV, Ob0=None, *, name=None, meta=None):
         super().__init__(H0=H0, Om0=Om0, Ode0=Ode0, Tcmb0=Tcmb0, Neff=Neff,
                          m_nu=m_nu, Ob0=Ob0, name=name, meta=meta)
-        self._wp = float(wp)
-        self._wa = float(wa)
-        self._zp = zp << self.__class__.zp.unit
+        self.wp = float(wp)
+        self.wa = float(wa)
+        self.zp = zp
```

Comment:
Here, could one use `fvalidate=aszarr`?

Reply:
Yes! that would be great. When https://github.com/astropy/astropy/pull/12169 is in ``fvalidate=aszarr`` is definitely the way to go.

Reply:
Haha. Mispoke. That strips the units from ``zp``. I like the units!
I did speed up some internal calculations using ``self._zp.value`` instead of ``self._zp`` which always works b/c ``self._zp`` is always in units of redshift.